source/common/router/context_impl.h

```diff
@@ -25,6 +25,7 @@ namespace Router {
   COUNTER(rq_redirect)                                                                             \
   COUNTER(rq_reset_after_downstream_response_started)                                              \
   COUNTER(rq_total)                                                                                \
+  HISTOGRAM(upstream_service_time, Milliseconds)                                                   \
```

Comment:
nit: to align with [current naming conventions](https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats#config-cluster-manager-cluster-stats) for upstream stats, this can be named smth like `upstream_rq_processing_ms`